drivers/platform/surface/surface_platform_profile.c

```diff
@@ -42,6 +52,13 @@ SSAM_DEFINE_SYNC_REQUEST_CL_W(__ssam_tmp_profile_set, __le32, {
 	.command_id      = 0x03,
 });
 
+SSAM_DEFINE_SYNC_REQUEST_W(__ssam_fan_profile_set, char, {
+	.target_category = SSAM_SSH_TC_FAN,
+	.target_id = SSAM_SSH_TID_SAM,
+	.command_id = 0x0e,
+	.instance_id = 0x01,
+});
```

Comment:
I couldn't use the `SSAM_DEFINE_SYNC_REQUEST_CL_R` here, as that would send commands to the instance id of the `TMP` module, this works, but I'm not sure if it's the best way to solve that.

Reply:
Hmm, that is actually a good question. Windows uses IID 1, correct?

This also kind of begs the question how multiple fans would be handled here. I.e., whether each would have its own fan profile or whether there is a single fan profile for all fans... If there would be only a single profile, IID 0 should be preferred, otherwise the IID of the respective fan.

Unfortunately, that's not something that we can easily answer I think (if at all). So I guess for now go with whatever Windows uses.

Reply:
Yeah, windows uses instance id 1, from the description;
```
2023-12-16 6:54:41 PM   Seq OUT FAN:05 t01 i01 c 0x0e (  1): 02 
```
`i01` [stands for](https://github.com/iwanders/nixos-surface/blob/a3196593eb51deef04f6e5152f5871b048ace30a/dev/irp_display.py#L274-L286) instance id 01.

I guess we'll find out how it works with multiple fans if they ever make a surface with two fans? :D  Is the surface pro 9 the first one with a fan?, or have others had fans but no one really dug into handling them?

Should I explore whether sending to instance id 0 works? Is that the 'apply all all instances' flag?

Reply:
> I guess we'll find out how it works with multiple fans if they ever make a surface with two fans? :D Is the surface pro 9 the first one with a fan?, or have others had fans but no one really dug into handling them?

I suspect that the SLS and some other of the new devices also have fan control support (an easy way to check seems to be the ACPI PNP-whatever FAN node). The SB2 has a fan in the base for the dGPU, but all I found for controlling that was the performance profile. So on the newer devices I think just no one looked into this before you.

> Should I explore whether sending to instance id 0 works? Is that the 'apply all all instances' flag?

I think it's probably best to stick with whatever windows does for now.